samples/snippets/src/main/java/com/examples/documentai/BatchParseForm.java

```diff
@@ -0,0 +1,179 @@
+      String parent = String.format("projects/%s/locations/%s", projectId, location);
+
+      KeyValuePairHint keyValuePairHint =
+          KeyValuePairHint.newBuilder().setKey("Phone").addValueTypes("PHONE_NUMBER").build();
+
+      KeyValuePairHint keyValuePairHint2 =
+          KeyValuePairHint.newBuilder()
+              .setKey("Contact")
+              .addValueTypes("EMAIL")
+              .addValueTypes("NAME")
+              .build();
+
+      FormExtractionParams params =
+          FormExtractionParams.newBuilder()
+              .setEnabled(true)
+              .addKeyValuePairHints(keyValuePairHint)
+              .addKeyValuePairHints(keyValuePairHint2)
+              .build();
+
+      GcsSource inputUri = GcsSource.newBuilder().setUri(inputGcsUri).build();
+
+      InputConfig config =
+          InputConfig.newBuilder().setGcsSource(inputUri).setMimeType("application/pdf").build();
+
+      ProcessDocumentRequest req =
+          ProcessDocumentRequest.newBuilder()
+              .setFormExtractionParams(params)
+              .setInputConfig(config)
+              .setOutputConfig(
+                  OutputConfig.newBuilder()
+                      .setGcsDestination(
+                          GcsDestination.newBuilder()
+                              .setUri(
+                                  String.format("gs://%s/%s", outputGcsBucketName, outputGcsPrefix))
+                              .build())
+                      .setPagesPerShard(1)
+                      .build())
+              .build();
+
+      BatchProcessDocumentsRequest requests =
+          BatchProcessDocumentsRequest.newBuilder().addRequests(req).setParent(parent).build();
+
+      // Batch process document using a long-running operation.
+      OperationFuture<BatchProcessDocumentsResponse, OperationMetadata> future =
+          client.batchProcessDocumentsAsync(requests);
+
+      // Wait for operation to complete.
+      System.out.println("Waiting for operation to complete...");
+      future.get(300, TimeUnit.SECONDS);
+
+      System.out.println("Document processing complete.");
+
+      Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+      Bucket bucket = storage.get(outputGcsBucketName);
+
+      // List all of the files in the Storage bucket.
+      Page<Blob> blobs =
+          bucket.list(
+              Storage.BlobListOption.currentDirectory(),
+              Storage.BlobListOption.prefix(outputGcsPrefix));
+
+      int idx = 0;
+      for (Blob blob : blobs.iterateAll()) {
+        if (!blob.isDirectory()) {
+          System.out.printf("Fetched file #%d\n", ++idx);
+          // Read the results
+
+          // Download and store json data in a temp file.
+          File tempFile = File.createTempFile("file", ".json");
+          Blob fileInfo = storage.get(BlobId.of(outputGcsBucketName, blob.getName()));
+          fileInfo.downloadTo(tempFile.toPath());
```

Comment:
Instead of creating a temp file, could you use `blob.getContent()`?

Reply:
I tried using `          Document document = builder.setContent(ByteString.copyFrom(blob.getContent())).build();` However, it is not parsing correctly.

I tried other ways of parsing Document from byte array, but they were not parsing correctly.